text/0000-introduce empty ranges.md

```diff
@@ -0,0 +1,267 @@
+Range(inf, -inf, 10): .is_empty() = true, .is_infinite() = false
+Range(inf, -inf, inf): .is_empty() = true, .is_infinite() = false
+
+//Infinite Ranges
+
+Range(0, 10, inf): .is_empty() = false, .is_infinite() = true
+Range(0, -10, -inf): .is_empty() = false, .is_infinite() = true
+Range(0, 10, nan): .is_empty() = false, .is_infinite() = true
+Range(0, inf, 10): .is_empty() = false, .is_infinite() = true
+Range(0, inf, 0): .is_empty() = false, .is_infinite() = true
+Range(0, inf, inf): .is_empty() = false, .is_infinite() = true
+Range(0, -inf, 0): .is_empty() = false, .is_infinite() = true
+Range(0, -inf, -10): .is_empty() = false, .is_infinite() = true
+Range(0, -inf, -inf): .is_empty() = false, .is_infinite() = true
+Range(inf, 0, 0): .is_empty() = false, .is_infinite() = true
+Range(inf, 0, -10): .is_empty() = false, .is_infinite() = true
+Range(inf, 0, -inf): .is_empty() = false, .is_infinite() = true
+Range(-inf, 0, 0): .is_empty() = false, .is_infinite() = true
+Range(-inf, 0, 10): .is_empty() = false, .is_infinite() = true
+Range(-inf, 0, inf): .is_empty() = false, .is_infinite() = true
+Range(inf, 0, nan): .is_empty() = false, .is_infinite() = true
+Range(0, inf, nan): .is_empty() = false, .is_infinite() = true
+Range(-inf, 0, nan): .is_empty() = false, .is_infinite() = true
+Range(0, -inf, nan): .is_empty() = false, .is_infinite() = true
+Range(nan, 0, inf): .is_empty() = false, .is_infinite() = true
+Range(0, nan, inf): .is_empty() = false, .is_infinite() = true
+Range(nan, 0, -inf): .is_empty() = false, .is_infinite() = true
+Range(0, nan, -inf): .is_empty() = false, .is_infinite() = true
+Range(inf, inf, 1): .is_empty() = false, .is_infinite() = true
+Range(inf, inf, -1): .is_empty() = false, .is_infinite() = true
+Range(inf, inf, 0): .is_empty() = false, .is_infinite() = true
+Range(inf, inf, nan): .is_empty() = false, .is_infinite() = true
+Range(inf, inf, inf): .is_empty() = false, .is_infinite() = true
+Range(inf, inf, -inf): .is_empty() = false, .is_infinite() = true
+Range(-inf, -inf, 1): .is_empty() = false, .is_infinite() = true
+Range(-inf, -inf, -1): .is_empty() = false, .is_infinite() = true
+Range(-inf, -inf, 0): .is_empty() = false, .is_infinite() = true
+Range(-inf, -inf, nan): .is_empty() = false, .is_infinite() = true
+Range(-inf, -inf, inf): .is_empty() = false, .is_infinite() = true
+Range(-inf, -inf, -inf): .is_empty() = false, .is_infinite() = true
+Range(nan, nan, 0): .is_empty() = false, .is_infinite() = true
+Range(nan, nan, 1): .is_empty() = false, .is_infinite() = true
+Range(nan, nan, inf): .is_empty() = false, .is_infinite() = true
+Range(nan, nan, -inf): .is_empty() = false, .is_infinite() = true
+```
+
+# How We Teach This
+
+This can be taught by improving the documentation of `Range` and by discussing a few examples in the tutorial section. Also, as stated above, `Range.is_infinite()` is hardly used on Github and the impact of this change should be considered mild.
```

Comment:
Please detail how the documentation would be changed in as much detail as makes sense.